app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -1024,14 +1004,31 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
                     if (!lastValue.toString().equals(getString(chooseDateOptionStringId))) {
                         dateField.setText(lastValue);
                     }
-                    DialogFragment datePickerFragment = DatePickerFragment.newInstance(
-                            loyaltyCardField,
-                            (Date) dateField.getTag(),
```

Comment:
We should set the current date of the picker to the current value of the field (if any), thus we need to pass this value to `showDatePicker()` like it was passed to `DatePickerFragment.newInstance()` (which took a `currentDate` parameter) as well.

Reply:
Done